source/common/upstream/upstream_impl.h

```diff
@@ -704,8 +705,9 @@ class ClusterInfoImpl : public ClusterInfo,
                   Stats::ScopeSharedPtr&& stats_scope, bool added_via_api,
                   Server::Configuration::TransportSocketFactoryContext&);
 
-  static LazyClusterTrafficStats generateStats(Stats::Scope& scope,
-                                               const ClusterTrafficStatNames& cluster_stat_names);
+  static std::unique_ptr<LazyableClusterTrafficStats>
+  generateStats(Stats::Scope& scope, const ClusterTrafficStatNames& cluster_stat_names,
+                bool lazyinit = false);
```

Comment:
I feel like we should remove the default and force all call-sites to determine the enable-lazy setting.